addons/compat_ace_fire/functions/fnc_burnSimulation.sqf

```diff
@@ -0,0 +1,193 @@
+                    };
+
+                    _intensity = _intensity - (1 / _intensity);
+                } else {
+                    // Make the unit leave the vehicle
+                    if (_vehicle != _unit) then {
+                        //TRACE_1("Ejecting",_unit);
+
+                        _unit leaveVehicle _vehicle;
+                        unassignVehicle _unit;
+
+                        _unit action ["Eject", _vehicle];
+                    };
+
+                    _unit disableAI "TARGET";
+                    _unit disableAI "AUTOTARGET";
+
+                    // Run away, erraticly
+                    if (leader group _unit != _unit) then {
+                        [_unit] join grpNull;
+                    };
+
+                    _unit doMove ((getPosATL _unit) getPos [20 + random 35, floor (random 360)]);
+                    _unit setSpeedMode "FULL";
+                    _unit setSuppression 1;
+                };
+            };
+
+            // Play screams and throw weapon (if enabled)
+            _unit call ace_fire_fnc_burnReaction;
+        };
+
+        // Keep pain around unconsciousness limit to allow for more fun interactions
+        private _painPercieved = (0 max ((_unit getVariable ["ace_medical_pain", 0]) - (_unit getVariable ["ace_medical_painSuppress", 0])) min 1);
+        private _painUnconscious = missionNamespace getVariable ["ace_medical_painUnconsciousThreshold", 0];
+        private _damageToAdd = [0.15, _intensity / BURN_MAX_INTENSITY] select (!alive _unit || {_painPercieved < _painUnconscious + random 0.2});
+
+        if (missionNamespace getVariable ["ace_medical_enabled", false]) then {
+            if (!isNull _instigator) then {
+                _unit setVariable ["ace_medical_lastDamageSource", _instigator];
+                _unit setVariable ["ace_medical_lastInstigator", _instigator];
+            };
+
+            // Common burn areas are the hands and face https://www.ncbi.nlm.nih.gov/pubmed/16899341/
+            private _bodyPart = ["Head", "Body", "LeftArm", "RightArm", "LeftLeg", "RightLeg"] selectRandomWeighted [0.77, 0.5, 0.8, 0.8, 0.3, 0.3];
+
+            if (!GVAR(ignoreArmorACE) && {alive _unit}) then {_damageToAdd = [_unit,  _damageToAdd, _damageToAdd, _unit, "", (_bodyPart isEqualTo "Body")] call EFUNC(main,receiveDamageACE);};
+            // Use event directly, as ace_medical_fnc_addDamageToUnit requires unit to be alive
+            if (_damageToAdd > 0) then {
+                ["ace_medical_woundReceived", [_unit, [[_damageToAdd, _bodyPart, _damageToAdd]], _instigator, "burn"]] call CBA_fnc_localEvent;
+            };
+        } else {
+            private _bodyParts = [["HitFace", "HitNeck", "HitHead"], ["HitPelvis", "HitAbdomen", "HitDiaphragm", "HitChest", "HitBody"], ["HitArms", "HitHands"], ["HitLegs"]] selectRandomWeighted [0.77, 0.5, 0.8, 0.3];
+            _damageToAdd = _damageToAdd * GVAR(fireMult);
+
+            if (EGVAR(main,enable)) exitWith {
```

Comment:
`if ... then ... else` preferred here. Basically already what this does, but makes more readable